src/layout-rect.js

```diff
@@ -280,3 +280,28 @@ export function cloneLayoutMarginsChangeDef(marginsChange) {
     right: marginsChange.right,
   };
 }
+
+/**
+ * Return position of element relative to top-left of document. Uses
+ * getBoundingClientRect(), which is expensive
+ * @param {Object} element
+ * @param {Object} hostViewport
+ * @return {Object}
+ */
+export function getClientRectRelativeDoc(element, hostViewport) {
+  const rect = element./*OK*/ getBoundingClientRect();
```

Comment:
`getBoundingClientRect()` can be quite expensive. Let's batch the measurement into the `vsync.measurePromise`. 
Note this makes the caller function async, but it should be fine for InOb Polyfill. Because native InOb batches multiple entires and fire asynchronously. 

You can refer to the Viewport method here. https://github.com/ampproject/amphtml/blob/master/src/service/viewport/viewport-impl.js#L377